skills/csharp/experimental/restaurantbookingskill/Dialogs/MainDialog.cs

```diff
@@ -181,16 +181,23 @@ protected async Task<bool> InterruptDialogAsync(DialogContext innerDc, Cancellat
 
                                 await innerDc.Context.SendActivityAsync(_localeTemplateManager.GenerateActivity(RestaurantBookingSharedResponses.CancellingMessage));
                                 await innerDc.CancelAllDialogsAsync();
-                                await innerDc.BeginDialogAsync(InitialDialogId);
-                                interrupted = true;
+                                if (innerDc.Context.IsSkill())
+                                {
+                                    interrupted = await innerDc.EndDialogAsync(cancellationToken: cancellationToken);
```

Comment:
should return action result here.
see https://github.com/microsoft/botframework-skills/blob/5612be74c909ba40e10b5c88b35334c2b1efd3c0/skills/csharp/pointofinterestskill/Dialogs/MainDialog.cs#L187